### PR TITLE
EWSGhana: Changed SOH tests

### PR DESCRIPTION
**Pull request by kkrampa**
@czue 
Now responses in v1 and in HQ are exactly the same. Mike wants to verify scripts so it's easier when messages are the same.

### CODE REVIEW

File: custom/ewsghana/handlers/alerts.py [hunk 1]
```diff
@@ -5,10 +6,173 @@
...
+                  or mylist[i].isalpha() and not mylist[i + 1].isalpha()
+                  or not mylist[i].isalnum() and mylist[i + 1].isalnum()):
+                yield token
+                token = ''
+            i += 1
+
+    def format(self, string):
+        """
+        Old parse method, used in Ghana for more 'interesting' parsing.
+        Moved from: https://github.com/dimagi/rapidsms-logistics/blob/7a1433abbda4ec27dc8f4c5da14c0f5689abd202/logistics/models.py#L1430
```
Comment (dimagimon) [inline custom/ewsghana/handlers/alerts.py]: E501 line too long (136 > 115 characters)

File: custom/ewsghana/handlers/alerts.py [hunk 2]
```diff
@@ -5,10 +6,173 @@
...
 from corehq.apps.sms.api import send_sms_to_verified_number
 from corehq.apps.commtrack.sms import *
+from custom.ilsgateway.tanzania.reminders import SOH_HELP_MESSAGE
+
+
+class ProductCodeException(Exception):
+    pass
+
+
+class EWSFormatter(object):
```
Comment (kkrampa) [inline custom/ewsghana/handlers/alerts.py]: Code moved from v1

File: custom/ewsghana/handlers/alerts.py (hunk 1 above)
Comment (kkrampa) [inline custom/ewsghana/handlers/alerts.py]: It's just url. I don't want to break it.

File: custom/ewsghana/handlers/alerts.py [hunk 3]
```diff
@@ -5,10 +6,173 @@
...
+    def single_action_transactions(self, action, args):
+        products = []
+        for idx, arg in enumerate(args):
+            if self.looks_like_prod_code(arg):
+                product = self.product_from_code(arg)
+                if product:
+                    products.append(product)
+                else:
+                    if idx == 0:
+                        raise ProductCodeException(INVALID_PRODUCT_CODE % arg)
```
Comment (kkrampa) [inline custom/ewsghana/handlers/alerts.py]: In EWS when first product is invalid exception is raised and other products aren't processed.

File: custom/ewsghana/handlers/alerts.py [hunk 4]
```diff
@@ -5,10 +6,173 @@
...
+                    self.bad_codes.add(arg)
+            else:
+                if not products:
```
Comment (kkrampa) [inline custom/ewsghana/handlers/alerts.py]: When first product is valid and some other is invalid we skip that product.

File: custom/ewsghana/tests/handlers/utils.py [hunk 5]
```diff
@@ -113,9 +121,16 @@ def setUpClass(cls):
                 caption='receipts'
             )
         )
-        config.consumption_config = ConsumptionConfig(min_transactions=0, min_window=0, optimal_window=60)
+        config.consumption_config = ConsumptionConfig(use_supply_point_type_default_consumption=True)
```
Comment (kkrampa) [inline custom/ewsghana/tests/handlers/utils.py]: To simplify tests we use manual consumption (manual consumption is also used in v1 tests)

File: custom/ewsghana/handlers/alerts.py (hunk 4 above)
Comment (czue) [inline custom/ewsghana/handlers/alerts.py]: fancy

**Pull request merged**